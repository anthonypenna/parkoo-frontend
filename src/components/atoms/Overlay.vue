<template>
  <Fade>
    <div class="overlay">
      <slot />
    </div>
  </Fade>
</template>

<script lang="ts">
import Vue from "vue";
import Fade from "@/components/atoms/Fade.vue";

export default Vue.extend({
  name: "Overlay",
  components: { Fade },
});
</script>

<style lang="scss" scoped>
.overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #fff;
  z-index: 10;
}
</style>
