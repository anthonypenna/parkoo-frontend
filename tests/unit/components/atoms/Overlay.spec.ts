import { shallowMount } from '@vue/test-utils'
import Overlay from '@/components/atoms/Overlay.vue'

describe('<Overlay />', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(Overlay, { slots: { default: 'Hello world!' } })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
