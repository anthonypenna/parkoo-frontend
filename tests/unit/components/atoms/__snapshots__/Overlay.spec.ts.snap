// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overlay /> should render correctly 1`] = `
<fade-stub>
  <div class="overlay">Hello world!</div>
</fade-stub>
`;
